### PR TITLE
Add convenience functions for parsing the PQ index

### PR DESCRIPTION
**Pull request by PhilipBAdams**
- [ ] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [ ] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
- [ ] Should this result in any changes to our documentation, either updating existing docs or adding new ones?

#### What does this implement/fix? Briefly explain your changes.
User may need to parse index graph information into different format for different applications. This PR gives an API for getting the neighborhood of a given vector, and that vector's PQ and full-dimensional data.

### CODE REVIEW

**Comment by harsha-simhadri** [comment]
@gopalrs is working on a PR to make distance class (including PQ) a member of data store. Can this wait until that PR? If we merge it now, we would have to redo a bit. I am open to both. @gopalrs?

**Comment by PhilipBAdams** [comment]
If the interface is ok with you I would prefer to merge it now and update the implementation after Gopal's PR so it is simpler to add it to my tool. If you want to have interface change, we can wait for Gopal's PR

File: src/pq_flash_index.cpp
Comment (gopalrs) [inline src/pq_flash_index.cpp:1568]: Should we need to specify std::move() here or will return value optimization take care of move semantics?

File: include/pq_flash_index.h
Comment (gopalrs) [inline include/pq_flash_index.h:99]: Should get_neighbors return vector of uint32_t? Or are you trying to make this future-proof?

File: src/pq_flash_index.cpp
Comment (gopalrs) [inline src/pq_flash_index.cpp:1539]: VID should be in small letters?

**Review by gopalrs — APPROVED** [review]
Agree with @harsha-simhadri that we could hold this off until the datastore/graphstore PR is done, if it is not urgent.

File: src/pq_flash_index.cpp
Comment (ltan1ms) [reply src/pq_flash_index.cpp:1608]: yes, i think it might be more efficient than relying on return value optimization (as it depends on caller pattern and may not always be guaranteed to avoid copying)?

File: src/pq_flash_index.cpp
Comment (ltan1ms) [inline src/pq_flash_index.cpp:1572]: Since the pq'ed vector is always in memory (without changing location), do you think we can just return a pointer to the underlying uint8_t array to save some efficiency?

**Review by ltan1ms — APPROVED** [review]

File: src/pq_flash_index.cpp
Comment (PhilipBAdams) [reply src/pq_flash_index.cpp]: I don't think it's possible to make a vector in-place on an existing array without having very tricky rules about what happens if the vector is resized or the edited.

I replace it with call to the copy constructor in rvalue position, so it is guaranteed to be return-optimized

**Comment by PhilipBAdams** [comment]
Hi @harsha-simhadri, do you mind to re-review/merge? If I move your new `read_nodes` to be public, the change is much simpler than it was before

File: src/pq_flash_index.cpp
Comment (harsha-simhadri) [reply src/pq_flash_index.cpp]: @philipBAdams could you please make the change Gopal requests

Comment (PhilipBAdams) [reply src/pq_flash_index.cpp]: This method no longer exists since I use your `read_nodes` method to get the vector coordinates instead

**Pull request merged**
